src/Features/Cipher/KeyedCipherStream/index.tsx

```diff
@@ -8,68 +8,41 @@ import FormControl from 'react-bootstrap/FormControl';
 import ToggleButton from 'react-bootstrap/ToggleButton';
 import ToggleButtonGroup from 'react-bootstrap/ToggleButtonGroup';
 import { useFocusInput } from '../../../Hooks/FocusInput';
-import { useLocalStorage } from '../../../Hooks/LocalStorage';
 import './index.scss';
 
 interface Props {
   cipher: KeyedCipherStringBase;
+  conversion: number;
   id: string;
+  secret: string;
+  onClear: () => void;
+  onConversionChange: (value: number) => void;
+  onSecretChange: (value: string) => void;
+  onTextChange: (value: string) => void;
   prompt: JSX.Element | string;
-}
-
-interface SavedState {
-  conversion: number;
-  key: string;
   text: string;
 }
 
 function KeyedCipherStream(props: Props) {
   const inputRef = useFocusInput();
-  const [conversion, setConversion] = useState(2);
-  const [key, setKey] = useState('');
-  const [text, setText] = useState('');
-
-  useLocalStorage<SavedState>(
-    props.id,
-    (savedState) => {
-      if (savedState) {
-        setConversion(savedState.conversion);
-        setKey(savedState.key);
-        setText(savedState.text);
-      }
-    },
-    () => {
-      return {
-        conversion,
-        key,
-        text,
-      };
-    });
-
-  function onTextChanged(event: ChangeEvent<HTMLInputElement>) {
-    setText(event.currentTarget.value);
-  }
 
-  function onKeyChanged(event: ChangeEvent<HTMLInputElement>) {
-    setKey(event.currentTarget.value);
+  function onClearClick() {
+    props.onClear();
   }
 
-  function onConversionChanged(event: number) {
-    setConversion(event);
+  function onKeyChange(event: ChangeEvent<HTMLInputElement>) {
+    props.onSecretChange(event.currentTarget.value);
   }
 
-  function onClearClick(event: MouseEvent<HTMLButtonElement>) {
-    setConversion(2);
-    setKey('');
-    setText('');
+  function onTextChange(event: ChangeEvent<HTMLInputElement>) {
+    props.onTextChange(event.currentTarget.value);
   }
 
   function calculateOutput() {
-    props.cipher.key = key;
-    props.cipher.text = text;
-    return conversion === 1 ?
-      props.cipher.encrypt() :
-      props.cipher.decrypt();
+    const cipher = props.cipher;
+    cipher.key = props.secret;
+    cipher.text = props.text;
+    return props.conversion === 1 ? cipher.encrypt() : cipher.decrypt();
```

Comment:
Not new in your change, but this feels like a magic number to me - should we have a constant for this?  Should it just be a boolean?